partials support

avalez opened the pull request:
I could not understand how partials are supposed to work and developed this fix to let the following work with jQuery.Tmpll:

context.partial('main', {}, {'main': 'template.html', 'part': 'template/part.hml'})

Fix has two things, support for above notation in sammy.js and support for partials in sammy.tmpl.js

avalez committed:
partials support, e.g. context.partial('main', {}, {'main': 'template/main.html', 'part': 'template/part.hml'}) but for jquery.tmpl only

endor commented:
it would be really nice if you could add a test for that. also it'd be great if you could split it into two commits: the first adding partials support to the sammy.js partial method and the second allowing partials in jquery.tmpl.

please explain why the change in the render method is necessary. the partials variable is already checked in loadPartials, though admittedly a check for typeof object would be more explicit than just a check if the variable has a truthy value. the code in the render method is much more concise the way it is now.

avalez committed:
Revert "partials support, e.g. context.partial('main', {}, {'main': 'template/main.html', 'part': 'template/part.hml'}) but for jquery.tmpl only"

This reverts commit 20619073186f8454d20bb88cf319d659e433f77b.

avalez committed:
Fix support for partials, e.g. for location handler:

  this.render('test/fixtures/item.tmpl', {name: 'avalez'}, function(html) {
    $('#content').html(html);
  }, {part: 'test/fixtures/partial.tmpl'}

avalez commented:
Hi Frank,

Thank you for looking at it, and for your suggestions, now I think I've found right fix, just for tmpl pluign, please check it.

It allows to do this in location handler:

```
  this.render('test/fixtures/item.tmpl', {name: 'avalez'}, function(html) {
    $('#content').html(html);
  }, {part: 'test/fixtures/partial.tmpl'});
```

But I thought that partials should also work like this:

```
    context.partials = {part: 'test/fixtures/partial.tmpl'};
    this.partial('test/fixtures/item.tmpl', {name: 'avalez'}, function(html) {
      $('#content').html(html);
    });
```

I would appreciate your help to make partials work like this if it's intended, and/or fix documentation for Sammy.Mustache, similarly to pull request #100.

Thank you.

endor commented:
thanks avalez! that commit looked pretty good. just had to fix a small thing which broke the plugin code if data wasn't set, which is the case in the example we're trying to get to work.

I checked the support for the syntax you mentioned, just as it is documented in the Sammy.Mustache plugin, and it works here as well:

```
    var app = $.sammy(function() {
      this.use('Tmpl', 'tmpl');

      this.get('#/hello/:name/to/:friend', function(context) {
        this.load('mypartial.tmpl')
            .then(function(partial) {
              context.partials = {hello_friend: partial};
              context.name = context.params.name;
              context.friend = context.params.friend;

              context.partial('mytemplate.tmpl');                 
            });
       });
    });

    $(function() {
      app.run()
    });
```

you are still correct, that the partial method itself does not accept partials.. I am not sure if it's necessary, but maybe it would be more consistent.

avalez commented:
Hi Frank,

Yes, it could be more consistent if partial method itself would accept partials :)

Also I can't get that example to work, but trying to pass you an example, it works there: http://jsfiddle.net/avalez/AyH5b/

Thank you for your great help!

P.S. Ah, your 'fix a small thing'  made this example to work - context.partials are not passed as parameter, but as 'this'
